### PR TITLE
Feature add warehouse ticket 17

### PR DESCRIPTION
**Pull request by Maedehshariifii**

### CODE REVIEW

**Review by ryn-ngn — DISMISSED** [review]

**Review by ryn-ngn — APPROVED** [review]

**Comment by rjtbansal** [comment]
When you create PR give some sort of description and also link the ticket to JIRA

File: src/components/Warehouse/AddWarehouse.js [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+    console.log(errors);
+    return errors;
+  };
+
+  //Function to handle form submission
+  const handleSubmit = async (e) => {
+    try {
+      e.preventDefault();
+      setErrors(validateForm(formData));
+      await axios.post("http://localhost:8080/api/warehouses", formData);
```
Comment (rjtbansal) [inline src/components/Warehouse/AddWarehouse.js:87]: You should be validating if errors exist before sending a POST request. 

I was able to submit an empty form. Even though the network request correctly returned 400 Bad Request, we should not even be sending an API call if validations dont pass. We save unnecessary network requests in the process

File: src/components/Warehouse/AddWarehouse.js [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+                <p className="form-error">
+                  <img
+                    src={ErrorIcon}
+                    alt="Error Icon"
+                    className="form-error__icon"
+                  />
+                  {` ${errors.address}`}
+                </p>
+              ) : null}
+              <br></br>
```
Comment (rjtbansal) [inline src/components/Warehouse/AddWarehouse.js:154]: This error handling is very redundant. You should just loop through the `errors` and dynamically fill the JSX.

**Review by rjtbansal — COMMENTED** [review]
Overall I am able to successfully add a warehouse but I would expect following behaviors in the form:
- As I type, validation should disappear once I have entered the correct information.
- Also since we are updating the state on every keystroke we should then have client side validation that shows up as I type and disappears once i have entered validation instead of filling entire form and getting errors only after form submission for good UX
- API errors from BE should appear in a modal
- UI is not correct for tablet mode. Its appearing like a mobile mode

**Pull request merged**